File: TP3/TP3/src/Ej9Again/Rectangle.java
package Ej9Again;

public class Rectangle extends Figure {
    private Point topLeft,bottomRight;

    public Rectangle(Point tL, Point bR){
        topLeft = tL;
        bottomRight = bR;
    }

    public double base(){return Math.abs(topLeft.getX() - bottomRight.getX()) ;}

    public double height(){return Math.abs(topLeft.getY() - bottomRight.getY());}

    public double area(){
        return base()*height();
    }

    public double perimeter(){
        return 2*base() +2*height();
    }

    @Override
    public String toString(){
        return "Rectangulo [%s , %s]".formatted(topLeft,bottomRight);

    }
}

File: TP8/EJ4/rectangle.rb
# frozen_string_literal: true
require_relative 'figure'
#require '../../TP8/TP3/point'

class Rectangle < Figure
  def initialize(top_left, bottom_right)
    @top_left = top_left
    @bottom_right = bottom_right
  end

  def base
    @bottom_right.x - @top_left.x
  end

  def height
    @bottom_right.y - @top_left.y
  end

  def area
    base * height
  end

  def perimeter
    2 * base + 2 * height
  end
  def to_s
    "Rectangle : [#{@bottom_right} , #{@top_left} ]"
  end
end
